api/all/src/main/java/io/opentelemetry/api/logs/LogRecordBuilder.java

```diff
@@ -66,9 +67,19 @@ public interface LogRecordBuilder {
   /** Set the severity text. */
   LogRecordBuilder setSeverityText(String severityText);
 
-  /** Set the body string. */
+  /**
+   * Set the body string.
+   *
+   * <p>Shorthand for calling {@link #setBody(AnyValue)} with {@link AnyValue#of(String)}.
+   */
   LogRecordBuilder setBody(String body);
```

Comment:
I think I'd rather see this string version have a default implementation that delegates to the AnyValue version, rather than the other way around. It seems safer and less potentially lossy that way.

Reply:
Unfortunately we can't do this because. Because of our backwards compatibility guarantees, any new interface method we add has to have a default implementation. The default implementation should do the most sensible thing possible, which in this case is to call the existing `setBody(String)` method with a string encoding of the anyvalue.

Reply:
> Because of our backwards compatibility guarantees, any new interface method we add has to have a default implementation.

Note that these are the rules we've been abiding by, but we decided at the spec level that this type of compatibility isn't actually required. It allows alternative API implementations to continue to be able to compile when new API versions come out, but we decided its reasonable to say that alternative API implementations should have to stay up to date with API releases. 

See this section of the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#extending-apisdk-abstractions

I'm inclined to continue with the practices we've been following unless doing so causes some egregious API user experience.